### PR TITLE
allow request's timeout and open timeout to be configurable

### PR DESCRIPTION
**Pull request by okhwaja**
related to feature request

https://github.com/mapbox/mapbox-sdk-rb/issues/54

### CODE REVIEW

**Comment by samfader** [comment]
@okhwaja Hmmm. I'm not sure either. Are you able to run them locally?

**Comment by okhwaja** [comment]
hey @samfader , yes it works locally for me
```
MapboxAccessToken=xxxxxx bundle exec ruby -Itest test/all_tests.rb
[Coveralls] Set up the SimpleCov formatter.
[Coveralls] Using SimpleCov's default settings.
Loaded suite test/all_tests
Started
...........................
Finished in 8.196894 seconds.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
27 tests, 34 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
3.29 tests/s, 4.15 assertions/s
[Coveralls] Outside the CI environment, not sending data.
```

**Comment by samfader** [comment]
I'm not quite sure why they aren't running on this repo. This repo doesn't really have a maintainer at this point so it's tricky. 

@andrii-rymar I know you've contributed somewhat recently here, would you be willing to be a third set of eyes on this PR?

File: test/geocoding_test.rb [hunk 1]
```diff
@@ -5,49 +5,54 @@
...
+    def teardown
+      Mapbox.request_open_timeout = nil
+      Mapbox.request_timeout = nil
+    end
+
+    should "#geocode_forward" do
       result = Mapbox::Geocoder.geocode_forward("Chester, NJ")
       assert result
+      assert Mapbox.request_opts[:timeout] = 80
+      assert Mapbox.request_opts[:open_timeout] = 30
```
Comment (andrii-rymar) [inline test/geocoding_test.rb:22]: Maybe extract these asserts to separate case without verifying the `result` as we are interested only if parameters set correctly? Like `should "set_default_timeouts"`

File: test/geocoding_test.rb [hunk 2]
```diff
@@ -66,5 +70,13 @@ class GeocodingTest < Test::Unit::TestCase
       end
     end
 
+    should "#geocode_forward (set request and open timeouts)" do
+      Mapbox.request_timeout = 10
+      Mapbox.request_open_timeout = 5
+      result = Mapbox::Geocoder.geocode_forward("Chester, NJ", {:proximity => {:longitude => 0.1234567, :latitude => -10.987654}})
+      assert result
```
Comment (andrii-rymar) [inline test/geocoding_test.rb:77]: Same thing - we are interested only in timeouts so no need to verify the result (such check already exists). So maybe just delete this row and change title to something like `should "set_custom_timeouts"`?

**Review by andrii-rymar — CHANGES_REQUESTED** [review]

**Comment by andrii-rymar** [comment]
@okhwaja @samfader added few comments but they are totally optional and it looks good enough to me even in the present shape.

File: test/geocoding_test.rb (hunk 1 above)
Comment (okhwaja) [reply test/geocoding_test.rb:26]: 👍 , addressed

**Comment by okhwaja** [comment]
hey @andrii-rymar @samfader , made the changes suggested in the comments. Verified the test suite still passes

```
MapboxAccessToken=xxxxxx bundle exec ruby -Itest test/all_tests.rb
[Coveralls] Set up the SimpleCov formatter.
[Coveralls] Using SimpleCov's default settings.
Loaded suite test/all_tests
Started
............................
Finished in 1.331453 seconds.
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
28 tests, 33 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
21.03 tests/s, 24.78 assertions/s
```

**Comment by andrii-rymar** [comment]
@okhwaja LGTM

@samfader what do you think?

**Review by andrii-rymar — APPROVED** [review]

**Comment by okhwaja** [comment]
gentle ping @samfader 😅 , just wanna be sure this doesn't fall off your radar

for context - we had an issue a few weeks ago where requests to mapbox were hanging and we're hoping to configure some timeouts to reduce the blast radius on our systems

**Comment by samfader** [comment]
@okhwaja Here it is: https://rubygems.org/gems/mapbox-sdk/versions/2.3.3

**Comment by okhwaja** [comment]
thanks @samfader !!

**Pull request merged**